internal/service/project/resource_project_test.go

```
@@ -1,20 +1,441 @@
+
+			if (err != nil) != tc.expectedError {
+				t.Errorf("Case %s: Received unexpected error: %v", tc.name, err)
+			}
+		})
+	}
+}
+
+func TestFilterUserDefinedLimits(t *testing.T) {
+	testCases := []struct {
+		name           string
+		allAtlasLimits []admin.DataFederationLimit
+		tfLimits       []project.TfLimitModel
+		expectedResult []admin.DataFederationLimit
+	}{
+		{
+			name: "FilterUserDefinedLimits",
+			allAtlasLimits: []admin.DataFederationLimit{
+				createDataFederationLimit("1"),
+				createDataFederationLimit("2"),
+				createDataFederationLimit("3"),
+			},
+			tfLimits: []project.TfLimitModel{
+				{
+					Name: types.StringValue("1"),
+				},
+				{
+					Name: types.StringValue("2"),
+				},
+			},
+			expectedResult: []admin.DataFederationLimit{
+				createDataFederationLimit("1"),
+				createDataFederationLimit("2"),
+			},
+		},
+		{
+			name: "FilterUserDefinedLimits",
+			allAtlasLimits: []admin.DataFederationLimit{
+				createDataFederationLimit("1"),
+			},
+			tfLimits:       []project.TfLimitModel{},
+			expectedResult: []admin.DataFederationLimit{},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			resultModel := project.FilterUserDefinedLimits(tc.allAtlasLimits, tc.tfLimits)
+			if !reflect.DeepEqual(resultModel, tc.expectedResult) {
+				t.Errorf("Filtered DataFederationlimit did not match expected output")
+			}
+		})
+	}
+}
+
+func TestUpdateProject(t *testing.T) {
+	testCases := []struct {
+		name          string
+		mockResponses []ProjectResponse
+		projectState  project.TfProjectRSModel
+		projectPlan   project.TfProjectRSModel
+		expectedError bool
+	}{
+		{
+			name:         "Successful update",
+			projectState: projectStateName,
+			projectPlan:  projectStateNameDiff,
+			mockResponses: []ProjectResponse{
+				{
+					Err: nil,
+				},
+			},
+			expectedError: false,
+		},
+		{
+			name:         "Same project names; Failed update",
```

why is this a failed update? seeing no error in mock response or expected errors

---

it's not a failed update, it's just that because the name is the same, it's not even trying to update. I will change the test case name to be more clear